shared/actions/chat/index.js

```diff
@@ -138,66 +143,67 @@ function* _incomingMessage(action: Constants.IncomingMessage): SagaGenerator<any
         const chatTabSelected = selectedTab === chatTab
         const conversationIsFocused =
           conversationIDKey === selectedConversationIDKey && appFocused && chatTabSelected && userActive
-        const {type: msgIDType, msgID: rpcMessageID} = Constants.parseMessageID(message.messageID)
-
-        if (message && message.messageID && conversationIsFocused && msgIDType === 'rpcMessageID') {
-          // TODO does this have to be sync?
-          try {
-            yield call(ChatTypes.localMarkAsReadLocalRpcPromise, {
-              param: {
-                conversationID: incomingMessage.convID,
-                msgID: rpcMessageID,
-              },
-            })
-          } catch (err) {
-            console.log(`Couldn't mark as read ${conversationIDKey} ${err}`)
+        if (message.messageID && conversationIsFocused) {
+          const {type: msgIDType, msgID: rpcMessageID} = Constants.parseMessageID(message.messageID)
+          if (msgIDType === 'rpcMessageID') {
+            // TODO does this have to be sync?
+            try {
+              yield call(ChatTypes.localMarkAsReadLocalRpcPromise, {
+                param: {
+                  conversationID: incomingMessage.convID,
+                  msgID: rpcMessageID,
+                },
+              })
+            } catch (err) {
+              console.log(`Couldn't mark as read ${conversationIDKey} ${err}`)
+            }
           }
         }
 
         const messageFromYou =
           message.deviceName === yourDeviceName && message.author && yourName === message.author
 
-        let pendingMessage
         if (
           (message.type === 'Text' || message.type === 'Attachment') &&
           messageFromYou &&
           message.outboxID
         ) {
-          pendingMessage = yield select(Shared.messageOutboxIDSelector, conversationIDKey, message.outboxID)
-        }
+          const outboxID: Constants.OutboxIDKey = message.outboxID
```

Comment:
flow can't seem to infer that later uses of message.outboxID can't be null unless we do this

Reply:
flow is correct here. We're passing message to a function and there's no guarantees that it won't mutate outboxID so it could become null